boards/FSM/Src/App/App_Coolant.c

```diff
@@ -0,0 +1,85 @@
+#include <stdlib.h>
+#include <assert.h>
+
+#include "App_Coolant.h"
+#include "App_InRangeCheck.h"
+#include "configs/App_FlowRateThresholds.h"
+#include "configs/App_SignalCallbackDurations.h"
+
+#include "App_SharedSetPeriodicCanSignals.h"
+STATIC_DEFINE_APP_SET_PERIODIC_CAN_SIGNALS_IN_RANGE_CHECK(FsmCanTxInterface)
+
+struct Coolant
+{
+    float (*get_flow_rate)(void);
+    struct InRangeCheck *flow_rate_in_range_check;
+
+    float (*get_temperature_A)(void);
+    float (*get_temperature_B)(void);
+    float (*get_pressure_A)(void);
+    float (*get_pressure_B)(void);
+
+    struct Signal *flow_in_range;
+};
+
+struct Coolant *App_Coolant_Create(
+    float (*get_flow_rate)(void),
+    float (*get_temperature_A)(void),
+    float (*get_temperature_B)(void),
+    float (*get_pressure_A)(void),
+    float (*get_pressure_B)(void))
+{
+    struct Coolant *coolant = malloc(sizeof(struct Coolant));
+    assert(coolant != NULL);
+
+    // Flow rate
+    coolant->get_flow_rate = get_flow_rate;
+    coolant->flow_rate_in_range_check =
+        App_InRangeCheck_Create(get_flow_rate, MIN_FLOW_RATE_L_PER_MIN, MAX_FLOW_RATE_L_PER_MIN);
+    // Temperature
+    coolant->get_temperature_A = get_temperature_A;
+    coolant->get_temperature_B = get_temperature_B;
+    // Pressure
+    coolant->get_pressure_A = get_pressure_A;
+    coolant->get_pressure_B = get_pressure_B;
+
+    coolant->flow_in_range = App_SharedSignal_Create(FLOW_METER_ENTRY_HIGH_MS, FLOW_METER_EXIT_HIGH_MS);
+
+    return coolant;
+}
+
+void App_Coolant_Broadcast(struct FsmCanTxInterface *can_tx, struct Coolant *coolant, bool * coolantTriggerShutdown)
```

Comment:
Why not return the shutdown bool? Is there a reason for the bool as an arg?

Reply:
because in case we need to fish out more variables from inside the broadcast function, we can do it like this. cleaner and expandable.